### DISCUSSION TITLE
Linter rule to avoid calling RegExp constructor

### DISCUSSION THREAD

**Discussion by anonrig** [biomejs/biome#1405] [Ideas]
`new RegExp` should be avoided unless you know what you are doing, since it might cause really serious security vulnerabilities.

Example Eslint implementation: https://github.com/getsentry/sentry-javascript/pull/10009

**Comment by ematipico** [comment]
I read the PR, and still, I can't understand the security implications. Especially in `new RegEx` VS `/something/` when we both have "user's inputs". 

I am not discouraging the proposal, but we should understand the implications and provide good documentation to explain to the users why this rule exists.

**Reply by Conaclos** [reply]
Same here. This could be nice to have some pointer.

Using dynamic regex construction has certainly some performance cost. Thus, this could be a `performance` rule. Otherwise we could introduce a new group named `retsriction` (Similarly to the Clippy `restriction` group: `code that is not "bad", but may be useful to prevent`)

**Reply by anonrig** [reply]
Let me try to explain it:

With `new RegEx()` you can create a dynamic regexp from a variable, such as `new RegExp('/${name}/')`, but with `/hello/` you can't. This simple fact is important since, we warn the developer and discourage them from using the constructor unless they know what they are doing.

**Reply by Conaclos** [reply]
So, no security risk?

**Reply by anonrig** [reply]
> So, no security risk?

The risk is the possibility of adding unsanitized user input into the arguments of RegExp
